### PR TITLE
改善对jpeg格式的支持，并避免数组越界

### PR DESCRIPTION
**Pull request by DarkOfTheMoon**

### CODE REVIEW

**Comment by HinTak** [comment]
Interestingly, the sample in #69 seems to need this too! Now I have got to have a look at your changes.

File: cajparser.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import os
 import struct
+import imagesize
```
Comment (HinTak) [inline cajparser.py:3]: I would suggest doing import only when needed (ie if the width height look obviously wrong).

File: HNParsePage.py [hunk 2]
```diff
@@ -14,7 +14,10 @@ def __init__(self, data, old_style=False):
         self.offset = 0
         def Text(self, code):
             try:
-                self.characters.append(bytes([self.data[self.offset+5],self.data[self.offset+4]]).decode("gbk"))
+                if len(self.data) > self.offset+5:
+                    self.characters.append(bytes([self.data[self.offset+5],self.data[self.offset+4]]).decode("gbk"))
```
Comment (HinTak) [inline HNParsePage.py:18]: Yes,  a lot of this code is work in progress. Probably should be totally rewritten at some point.

**Review by HinTak — APPROVED** [review]

**Comment by HinTak** [comment]
I'll merge it as is now, and keep a note for myself for those future improvements.

**Comment by HinTak** [comment]
@DarkOfTheMoon thanks a lot for making the effort fixing those issues.

**Comment by HinTak** [comment]
@DarkOfTheMoon I ended up changing both locations slightly differently; you can have a look.

**Comment by DarkOfTheMoon** [comment]
@HinTak I just noticed your reply today, sorry. For the problems encountered using caj2pdf, I am personally very happy to solve and share the code, and you are welcome.

**Pull request merged**